core/controllers/android_e2e_config.py

```diff
@@ -0,0 +1,301 @@
+
+from __future__ import absolute_import  # pylint: disable=import-only-modules
+from __future__ import unicode_literals  # pylint: disable=import-only-modules
+
+import os
+
+from constants import constants
+from core.controllers import acl_decorators
+from core.controllers import base
+from core.domain import exp_domain
+from core.domain import exp_services
+from core.domain import fs_services
+from core.domain import opportunity_services
+from core.domain import question_domain
+from core.domain import question_services
+from core.domain import rights_manager
+from core.domain import skill_domain
+from core.domain import skill_services
+from core.domain import state_domain
+from core.domain import story_domain
+from core.domain import story_services
+from core.domain import subtopic_page_domain
+from core.domain import subtopic_page_services
+from core.domain import topic_domain
+from core.domain import topic_fetchers
+from core.domain import topic_services
+from core.domain import user_services
+import feconf
+import python_utils
+
+
+class InitializeAndroidTestDataHandler(base.BaseHandler):
+    """Handler to initialize android specific structures."""
+
+    URL_PATH_ARGS_SCHEMAS = {}
+    HANDLER_ARGS_SCHEMAS = {'POST': {}}
+
+    @acl_decorators.open_access
+    def post(self):
+        """This method generates the following structures:
+            Topic: A topic with both a test story and a subtopic.
+            Story: A story with 'android_interactions' as a exploration
+                node.
+            Exploration: 'android_interactions' from the local assets.
+            Subtopic: A dummy subtopic to validate the topic.
+            Skill: A dummy skill to validate the subtopic.
+
+            All these structures are used for Android end-to-end testing, the
+            purpose is to test the network requests in android by fetching the
+            topic from the dev server.
+
+            This handler can only be used in the dev server, It gives a
+            500 error in the production server.
+
+            Returns a empty json - {}, when the request is successful.
+
+            This is only called once and raises a exception if the topic is
+            already created or published.
+
+        Raises:
+            Exception. When used in production mode.
+            Exception. The topic is already created but not published.
+            Exception. The topic is already published.
+        """
+
+        if not constants.DEV_MODE:
+            raise Exception('Cannot load new structures data in production.')
+        if topic_services.does_topic_with_name_exist(
+                'Android test'):
+            topic = topic_fetchers.get_topic_by_name('Android test')
+            topic_rights = topic_fetchers.get_topic_rights(
+                topic.id, strict=False)
+            if topic_rights.topic_is_published:
+                raise Exception('The topic is already published.')
```

Comment:
Ditto*

Reply:
I'm not sure what you meant here.
We need a 500 here right.

Reply:
We should raise 400 HTTP* error here, 500 is a server error and I don't think it's a server issue here, right? Also, the message should be user-facing and the user is not aware of the what's topic/publication here and I believe the handler is expected to generate topic data, right? If yes, then the error message should be around `Unable to generate dummy data as it already exists and published` or something around that? (not sure!)

Reply:
Ack. Done